lib/dialects/abstract/query-generator.js

```diff
@@ -1024,6 +1036,54 @@ class QueryGenerator {
     }
   }
 
+  composeQuery(composition) {
+    if (!(composition instanceof Composition)) {
+      composition = new Composition(composition);
+    }
+
+    const result = {};
+    const bind = [];
+    const bindParam = this.bindParam(bind);
+
+    const queryItems = composition.items.map(item => {
+      if (item instanceof Slot) {
+        // Really binds, returns only acompaning sql string with bind placeholder
+        return this.format(item.value, item.field, item.options, bindParam);
+      }
+
+      if (typeof item !== 'string') {
+        throw new sequelizeError.CompositionError('Item to be composed is not a slot or a string', item);
+      }
+
+      return item;
+    });
+
+    queryItems.push(';');
+    result.query = queryItems.join('');
```

Comment:
Nit: you can do `return { query: ..., bind: ... }` and remove the declaration of `result`.

Reply:
Done